app/src/main/java/com/example/home/preferences/PreferenceFragment.java

```diff
@@ -0,0 +1,127 @@
+package com.example.home.preferences;
+
+import android.content.Intent;
+import android.location.Geocoder;
+import android.os.Bundle;
+
+import androidx.constraintlayout.widget.ConstraintLayout;
+import androidx.fragment.app.Fragment;
+
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.EditText;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.example.home.MainActivity;
+import com.example.home.R;
+import com.example.home.login.LoginActivity;
+import com.example.home.stream.StreamFragment;
+import com.parse.ParseUser;
+
+import android.widget.AutoCompleteTextView;
+
+
+public class PreferenceFragment extends Fragment {
+    private ConstraintLayout mPrefConLayout;
+    private TextView mPreferenceTextView;
+    private Button mLogoutButton;
+    private AutoCompleteTextView mBedroomTextView;
+    private ArrayAdapter bedroomAdapter;
+    MainActivity mainActivity;
+    private AutoCompleteTextView mProperyType;
+    private ArrayAdapter propertyTypeAdapter;
+    private EditText mZipcodeEditText;
+    private Button mSaveButton;
+
+    private int mZipCode;
+    private String mZipCodeText;
+    private String mNoOfBedrooms;
+    private String mPropertyTypeText;
+
+    public PreferenceFragment(){
+    }
+
+    public PreferenceFragment(MainActivity mainActivity) {
```

Comment:
you shouldn't need this. A fragment by default has access to the parent activity